### ISSUE TITLE
Style/NegatedIfElseCondition with Style/TernaryParentheses

### ISSUE THREAD

**Issue by tsugimoto** [rubocop/rubocop#10924]
`Style/NegatedIfElseCondition` should look inside parantheses on ternary conditions because `Style/TernaryParentheses` may require them when the condition is complex.

Edit: I have managed to create an example with normal `if` when `Style/ParenthesesAroundCondition` allows parantheses with multiline conditions.

--------

## Expected behavior

Should detect offense.

## Actual behavior

No offenses detected.

#### main.rb

```rb
# frozen_string_literal: true

result = (user.coins != 0) ? :have_coins : :no_coins
puts result

if (user.coins !=
   0)
  puts 'have_coins'
else
  puts 'no_coins'
end
```

#### .rubocop.yml
```yml
AllCops:
  NewCops: enable

Style/TernaryParentheses:
  EnforcedStyle: require_parentheses_when_complex

Style/ParenthesesAroundCondition:
  AllowInMultilineConditions: true

Style/NumericPredicate: 
  Enabled: false
```

I have intentionally disabled `Style/NumericPredicate` in this example because default RuboCop prefers `zero?`, and I suppose it is not really related to this issue.

## RuboCop version

```
$ rubocop -V
1.35.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.1 x86_64-darwin20)
```
